Src/firmware/mainf.cpp

```diff
@@ -422,19 +526,32 @@ void update() {
   }
 
   if (battery_avg < params.battery_min_voltage) {
-    if (cnt % 10 == 0) gpio_toggle(LED);
+    battery_led_status = BatteryLedStatus::LOW_BATTERY;
   } else {
-    if (!ros_initialized) {
-      if (cnt % 50 == 0) gpio_toggle(LED);
+    if (status == AgentStatus::BOOT) {
+      battery_led_status = BatteryLedStatus::BOOT;
+    } else if (status != AgentStatus::AGENT_CONNECTED) {
+      battery_led_status = BatteryLedStatus::NOT_CONNECTED;
     } else {
-      gpio_reset(LED);
+      battery_led_status = BatteryLedStatus::CONNECTED;
     }
   }
 
-  if (!ros_initialized) return;
+  update_battery_led(cnt);
 
+  if (status == AgentStatus::BOOT) {
+    if (cnt % 500 == 0 && !publish_param_trigger) {
+      publish_param_trigger = true;
+    }
+
+    if (cnt % 2000 == 0 && !force_boot) {
+      force_boot = true;
+    }
```

Comment:
This is not a way of measuring 20 seconds as you don't know the starting cnt value when entering boot state

Reply:
Changed to usage of `time` method